src/bun.js/bindings/wtf-bindings.cpp

```diff
@@ -237,4 +237,21 @@ extern "C" size_t WTF__base64URLEncode(const unsigned char* __restrict inputData
         destinationDataBuffer[didx++] = '=';
 
     return destinationDataBufferSize;
-}
\ No newline at end of file
+}
+
+namespace Bun {
+String base64URLEncodeToString(Vector<uint8_t> data)
+{
+    auto size = data.size();
+    size_t encodedLength = ((size * 4) + 2) / 3;
+    if (!encodedLength)
+        return String();
+
+    Vector<uint8_t> encodedData(encodedLength);
+    encodedLength = WTF__base64URLEncode(data.data(), data.size(), encodedData.data(), encodedData.size());
+    if (!encodedLength)
+        return String();
+
+    return String::fromUTF8(encodedData.data(), encodedLength);
```

Comment:
We can avoid the extra clone here because it will always be latin1

LChar *ptr;
auto result = String::createUninitialized(ptr, encodedLength);
encodedLength = WTF__base64URLEncode(data.data(), data.size(), encodedData.data(), encodedData.size());
RELEASE_ASSERT(result.length() == encodedLength);
return result;